src/main/java/org/candlepin/subscriptions/subscription/SubscriptionSyncController.java

```diff
@@ -43,41 +49,67 @@
 @Component
 @Slf4j
 public class SubscriptionSyncController {
-  private static final int SUBS_PAGE_SIZE = 1000;
+  private static final int SUBS_PAGE_SIZE = 2000;
 
   private SubscriptionRepository subscriptionRepository;
   private OrgConfigRepository orgRepository;
+  private OfferingRepository offeringRepository;
   private SubscriptionService subscriptionService;
   private ApplicationClock clock;
   private CapacityReconciliationController capacityReconciliationController;
   private Timer syncTimer;
   private Timer enqueueAllTimer;
   private KafkaTemplate<String, SyncSubscriptionsTask> syncSubscriptionsByOrgKafkaTemplate;
   private String syncSubscriptionsTopic;
+  private final ProductWhitelist productWhitelist;
 
   @Autowired
   public SubscriptionSyncController(
       SubscriptionRepository subscriptionRepository,
       OrgConfigRepository orgRepository,
+      OfferingRepository offeringRepository,
       ApplicationClock clock,
       SubscriptionService subscriptionService,
       CapacityReconciliationController capacityReconciliationController,
       MeterRegistry meterRegistry,
       KafkaTemplate<String, SyncSubscriptionsTask> syncSubscriptionsByOrgKafkaTemplate,
+      ProductWhitelist productWhitelist,
       @Qualifier("syncSubscriptionTasks") TaskQueueProperties props) {
     this.subscriptionRepository = subscriptionRepository;
     this.orgRepository = orgRepository;
+    this.offeringRepository = offeringRepository;
     this.subscriptionService = subscriptionService;
     this.capacityReconciliationController = capacityReconciliationController;
     this.clock = clock;
     this.syncTimer = meterRegistry.timer("swatch_subscription_sync_page");
     this.enqueueAllTimer = meterRegistry.timer("swatch_subscription_sync_enqueue_all");
+    this.productWhitelist = productWhitelist;
     this.syncSubscriptionsTopic = props.getTopic();
     this.syncSubscriptionsByOrgKafkaTemplate = syncSubscriptionsByOrgKafkaTemplate;
   }
 
   @Transactional
   public void syncSubscription(Subscription subscription) {
+    String sku = sku(subscription);
+
+    if (!productWhitelist.productIdMatches(sku)) {
+      log.info(
+          "Sku {} not on allowlist, skipping subscription sync for subscriptionId: {} in org: {} ",
+          sku,
+          subscription.getId(),
+          subscription.getWebCustomerId());
+      return;
+    }
+
+    if (offeringRepository.findById(sku).isEmpty()) {
```

Comment:
Theoretically this can be more performant (Spring Data should generate a `select count` rather than fetching whole entity):

```suggestion
    if (!offeringRepository.existsById(sku)) {
```

Reply:
Makes sense. Fixed.